### PR TITLE
mortar shells will now cook off and explode when flamed

### PR DESCRIPTION
**Pull request by iloveloopers**
# About the pull request
title

# Explain why it's good for the game
immersion. ammo boxes do it, why not shells?
realism (probably)
its gonna be real funny on evac

# Testing Photographs and Procedure

https://github.com/cmss13-devs/cmss13/assets/140007537/a61e16dd-3119-4822-8ab2-8586895baddc

# Changelog

:cl:
add: Mortar shells will now cook off and explode when flamed
/:cl:

### CODE REVIEW

**Comment by vero5123** [comment]
You got a runtime buddy.
```

[2024-05-11 16:40:20.791] Starting up round ID 1
 - -------------------------)
[2024-05-11 16:40:20.814] Running CMSS13 revision: 2024-05-05
 - origin/master: 772d2f31ef35782affc8a6f2be553ca03d864c40
 - HEAD: 22e35ca812931d93912f23e9fb0ac222619f5999
[2024-05-11 17:24:17.210] runtime error: cell_explosion called without cause_data.
 - proc name: stack trace (/proc/stack_trace)
 -   source file: code/__HELPERS/unsorted.dm,1834
 -   usr: (src)
 -   src: null
 -   call stack:
 - stack trace("cell_explosion called without ...")
 - cell explosion(the plating (170,64,3) (/turf/open/floor/plating/almayer), 60, 20, 0, null, null)
 - the 80mm fragmentation mortar ... (/obj/item/mortar_shell/frag): detonate(the plating (170,64,3) (/turf/open/floor/plating/almayer))
 - /datum/callback (/datum/callback): InvokeAsync()
 - 

```

**Comment by vero5123** [comment]
There is also the issue of the incendiary mortars stacking fire damage. It's not uncommon for people to put all similar mortar shells into a pile and if the incendiary cooks off with multiple shells it looks like it stacks the damage when you walk over the fire, meaning an almost instant death depending on the number of shells.

File: code/modules/cm_marines/equipment/mortar/mortar_shells.dm [hunk 1]
```diff
@@ -154,6 +156,35 @@
 		icon_state = initial(icon_state) +"_unlocked"
 		playsound(loc, 'sound/items/Screwdriver2.ogg', 25, 0, 6)
 
+/obj/item/mortar_shell/ex_act(severity, explosion_direction)
+	if(!burning)
+		return ..()
+
+/obj/item/mortar_shell/attack_hand(mob/user)
+	if(burning)
+		to_chat(user, SPAN_DANGER("\The [src] is on fire and might explode!"))
```
Comment (Drulikar) [inline code/modules/cm_marines/equipment/mortar/mortar_shells.dm:165]:
```suggestion
		to_chat(user, SPAN_DANGER("[src] is on fire and might explode!"))
```

File: code/modules/cm_marines/equipment/mortar/mortar_shells.dm [hunk 2]
```diff
@@ -154,6 +156,35 @@
...
+
+/obj/item/mortar_shell/flamer_fire_act(dam, datum/cause_data/flame_cause_data)
+	if(burning)
+		return
+	burning = TRUE
+	cause_data = create_cause_data("Burning Mortar Shell", flame_cause_data.resolve_mob(), src)
+	handle_fire()
+
+/obj/item/mortar_shell/proc/handle_fire()
+	visible_message(SPAN_WARNING("\The [src] catches on fire and starts cooking off! It's gonna blow!"))
```
Comment (Drulikar) [inline code/modules/cm_marines/equipment/mortar/mortar_shells.dm:177]:
```suggestion
	visible_message(SPAN_WARNING("[src] catches on fire and starts cooking off! It's gonna blow!"))
```

**Review by Drulikar — APPROVED** [review]

**Comment by iloveloopers** [comment]
Man i can never get a perfect one can i?

**Pull request merged**